### PR TITLE
Continued sessions

### PR DESCRIPTION
**Pull request by galactoise**
Depends on #9, #10, and derpwizard #12 and #13

### CODE REVIEW

File: service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java [hunk 1]
```diff
@@ -88,17 +89,19 @@ public SpeechletResponseEnvelope doAlexaRequest(SpeechletRequestEnvelope request
     sessionAttributes.put("bot", "complibot");
 
     ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new MixInModule());
     QuipMetadata metadata = mapper.convertValue(sessionAttributes, new TypeReference<QuipMetadata>(){});
     
     SsmlDocumentBuilder builder = new SsmlDocumentBuilder(UNSUPPORTED_SSML_TAGS);
     VoiceInput voiceInput = VoiceMessageFactory.buildInputMessage(request.getRequest(), metadata, InterfaceType.ALEXA);
+    builder.conversationEnd(false);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java]: This is still unholy

File: service/src/main/java/com/derpgroup/quip/manager/QuipManager.java [hunk 2]
```diff
@@ -101,47 +119,56 @@ protected void doGoodbyeRequest(VoiceInput voiceInput,
...
       builder.text("Unknown request type '" + messageSubject + "'.");
     }
   }
 
-  private void doWhoIsRequest(VoiceInput voiceInput, SsmlDocumentBuilder builder) {
-    QuipMetadata metadata = (QuipMetadata) voiceInput.getMetadata();
+  private void doAnotherRequest(String messageSubject, Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
+    //this has its own method in case we want to do things like logging
+    ConversationHistoryEntry entry = ConversationHistoryUtils.getLastNonMetaRequestBySubject(metadata.getConversationHistory(), new HashSet<String>(Arrays.asList(metaRequestSubjects)));
+    switchOnSubject(entry.getMessageSubject(), messageMap, builder, metadata);
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/manager/QuipManager.java]: We need a null check here. If you ask for another without having a prior, it blows up.
"Alexa, ask Insultibot for another"

Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/manager/QuipManager.java]: This also blows up for help requests followed by "another"
"Alexa, ask insultibot for help"
[response]
"Another"
"unknown request type amazon.helpintent"

File: service/src/main/java/com/derpgroup/quip/manager/QuipManager.java [hunk 3]
```diff
@@ -101,47 +119,56 @@ protected void doGoodbyeRequest(VoiceInput voiceInput,
...
+      doInsultRequest(messageMap, builder, metadata);
       break;
     case "BACKHANDED_COMPLIMENT":
-      doBackhandedComplimentRequest(voiceInput, builder, metadata);
+      doBackhandedComplimentRequest(messageMap, builder, metadata);
       break;
     case "WINSULT":
-      doWinsultRequest(voiceInput, builder, metadata);
-      break;
-    case "HELP":
```
Comment (davidwphillips) [inline service/src/main/java/com/derpgroup/quip/manager/QuipManager.java]: We need HELP back in here as a switchable subject, else the "another" request can't trigger it to replay. And it blows up as:
"Unknown request type amazon.helpintent"

File: service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java (hunk 1 above)
Comment (galactoise) [inline service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java]: Sure is.

Comment (R-Gerard) [inline service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java]: Sloppy

**Pull request merged**